app/models/df_json_accessor.rb

```diff
@@ -0,0 +1,40 @@
+class DfJsonAccessor
+
+  def initialize(json)
+    @json = json
+  end
+
+  def self.define_json_accessors
+    self.selectors.each do |selector_name, selector_info|
+      selector_type = selector_info[:type]
+      case selector_type
+      when :money_amount
+        define_method(selector_name) do
+          df_json_value(__method__)&.to_d || 0
+        end
+      when :date
+        define_method(selector_name) do
+          Date.parse(df_json_value(__method__))
+        end
+      when :string, :boolean
+        define_method(selector_name) do
+          df_json_value(__method__)
+        end
+      else
+        define_method(selector_name) do
+          df_json_value(__method__)
+        end
+      end
+    end
+  end
+
+  def df_json_value(selector_name)
+    @json[selectors[selector_name][:key]]
```

Comment:
You may consider using 

```ruby
keys = selectors[selector_name][:key].split(' ')
@json.dig(*keys)
```

Which will enable deep traversal of the JSON structure without an API change.

You may or may not need to splat. You'll have to play with it to see. [Documentation linked for convenience](https://apidock.com/ruby/Hash/dig).

Reply:
awesome - that's exactly what i wanted and just got lazy and didn't look up how to do it/think of using dig :D